Server/controllers/auth.controller.js

```diff
@@ -1,13 +1,57 @@
 "use strict";
+const passport = require('passport');
+var jwt = require('jsonwebtoken');
+const passportConfig = require('../config/passport.config');
 /**
  * Class representing auth controller
  *
  * @class authController
  */
 class authController {
 
-    static login (req, res) {}
+  static login (req, res , next) {
+      passport.authenticate('local', function(err, user, info) {
+          if(err) {
+              return next(err);
+          }
+      
+          if (!user) { return res.json( { message: info.message }) }
+
+          const payload = {
+              username: user.username,
+              expires: Date.now() + parseInt(process.env.JWT_EXPIRATION_MS),
+            };
+      
+            /** assigns payload to req.user */
+            req.login(payload, {session: false}, (error) => {
+              if (error) {
+                res.status(400).send({ error });
+              }
+      
+              /** generate a signed json web token and return it in the response */
+              const token = jwt.sign(JSON.stringify(payload), "secretkey");
+              console.log(token);
+
+              // res.cookie('jwt', jwt, { httpOnly: true, secure: true });
+              // res.status(200).send({user});
```

Comment:
Remove console.log and commented out code

Reply:
Done